src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,24 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private Random random = new Random();
+    private Figure[] figures = {
+            new Circle(0),
+            new IsoscelesTrapezoid(0, 0, 0),
+            new Rectangle(0, 0),
+            new Square(0),
+            new RightTriangle(0, 0)};
```

Comment:
Don't use the constructor to pass in random values or expressions. Instead, generate random values within the `FigureSupplier` class and use them to create new instances of figures. This will improve the readability of your code and ensure that figures are not created with a size of 0, which doesn't make sense for a physical shape.